public/app/plugins/datasource/influxdb/components/VisualInfluxQLEditor/Editor.tags.test.tsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import { InfluxQuery } from '../../types';
+import InfluxDatasource from '../../datasource';
+import { render, screen, act } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { Editor } from './Editor';
+import * as mockedMeta from '../../influxQLMetadataQuery';
+
+jest.mock('../../influxQLMetadataQuery', () => {
+  // const orig = jest.requireActual('../../influxQLMetadataQuery');
+  return {
+    getTagKeysForMeasurementAndTags: jest
+      .fn()
+      // first time we are called when the widget mounts,
+      // we respond by saying `cpu, host` are the real tags
+      .mockReturnValueOnce(Promise.resolve(['cpu', 'host']))
+      // afterwards we will be called once when we click
+      // on a tag-key in the WHERE section.
+      // it does not matter what we return, as long as it is
+      // promise-of-a-list-of-strings
+      .mockReturnValueOnce(Promise.resolve([])),
+    getTagValues: jest
+      .fn()
+      // it does not matter what we return, as long as it is
+      // promise-of-a-list-of-strings
+      .mockReturnValueOnce(Promise.resolve([])),
+    getAllMeasurementsForTags: jest
+      .fn()
+      // it does not matter what we return, as long as it is
+      // promise-of-a-list-of-strings
+      .mockReturnValueOnce(Promise.resolve([])),
+  };
+});
+
+beforeEach(() => {
+  // we need to ts-ignore here, the functions in that module
+  // are mocked, so they do have `mockReset`, but typescript
+  // does not know this
+  // @ts-ignore
+  mockedMeta.getTagKeysForMeasurementAndTags.mockClear();
```

Comment:
Instead, you can also cast it with `as jest.Mock`

Reply:
sure